processing/src/main/java/io/druid/query/groupby/epinephelinae/column/DoubleGroupByColumnSelectorStrategy.java

```diff
@@ -46,7 +45,19 @@ public void processValueFromGroupingKey(
   @Override
   public void initColumnValues(ColumnValueSelector selector, int columnIndex, Object[] values)
   {
-   values[columnIndex] = ((DoubleColumnSelector) selector).get();
+   values[columnIndex] = selector.getDouble();
+  }
+
+  @Override
+  public Object getOnlyValue(ColumnValueSelector selector)
+  {
+    return selector.getDouble();
+  }
+
+  @Override
+  public void writeToKeyBuffer(int keyBufferPosition, Object obj, ByteBuffer keyBuffer)
+  {
+    keyBuffer.putDouble(keyBufferPosition, (Double) obj);
   }
 
```

Comment:
Maybe we can combine `getOnlyValue` and `writeToKeyBuffer` into a single method to avoid additional boxing and unboxing (similarly for `FloatGroupByColumnSelectorStrategy` and `LongGroupByColumnSelectorStrategy`)?

Please correct me if I am wrong but it seems that these two methods are always used together?